tock/tock/templates/base.html

```diff
@@ -7,7 +7,11 @@
     <title>Tock</title>
 
     {% load staticfiles %}
-    <link rel="shortcut icon" href="{% static 'img/favicon.ico' %}">
+    {% if x_tock_user_is_late %}
```

Comment:
I don't think you need a context processor for this. I think you can just do `{% if request.user.userdata.is_late %}

request.user _should_ always be available, and I think it's more clear to see the check than the cp-supplied variable,, which are always a little less transparent than one would like. I think the only advantage of a context processor would be if you were doing the check a lot to avoid extra DB lookups.

That said, I could see you wanting to add a "late" body class as a simple hook for some styling changes.

Reply:
Have you taken a look at the failing tests? Looks like not having timecard or reporting periods for all the users in all the tests is causing the tests to fail. But I’m not 100% sure either. But I do like that approach of checking the request object considering that’s where the user data lives in the first place.

Reply:
I haven't. But would the combo of skipping the CP and flipping the is_late check polarity clear it up?

Reply:
A user may not have all of the reporting periods or timecards associated with them though, so to me that seems to be an issue with the check itself.  Should we have a bit more error handling or checks in the code to look for the existence of the reporting period first before using it?

Then again, you're calling `.filter()` with them, not `.get()`, so I'm also not entirely sure why the tests are failing with a message saying `hours.models.DoesNotExist: ReportingPeriod matching query does not exist.`  Shouldn't that just be returning an empty QuerySet?